### ISSUE TITLE
AWS Parameter Store with Hoxton SR8+ doesn't override properties when using multiple profiles

### ISSUE THREAD

**Issue by peteguyatt** [spring-cloud/spring-cloud-aws#772]
**Type**: Bug

**Component**:

Parameter Store

**Describe the bug**
When using multiple active profiles with spring cloud Hoxton SR10 and boot 2.3.9 we have noticed that property order seems to have changed, for example if we have two properties set E.g.

spring.profiles.active=profile_a,profile_b

Parameter Store configured as:
```
application/profile_a/app.secret-key: abc-123
application/profile_b/app.secret-key: efd-213
```

and in our bean we have 
```java
@Value("${app.secret-key}")
private String secretKey; 
```

the value resolved will be abc-123 instead of efd-213

I looked through the commit history for ```AwsParamStorePropertySourceLocator``` and the issue seems to have been introduced in https://github.com/spring-cloud/spring-cloud-aws/commit/582fdc6691ead7904aaed176af2cf85221edacd5 where the ```Collections.reverse``` has been removed due to switching to using a ```LinkedHashSet```

This looks like an accidental bug to me and the same bug exists in master and the 2.3.x branch

**Comment by maciejwalkowiak** [comment]
@MatejNedic can you take a look at this one? We should fix it in both `2.2.x` in this repo and `2.3.x` in awspring.

**Comment by peteguyatt** [comment]
@WtfJoke You're correct the same issue exists with secrets manager too @maciejwalkowiak & @MatejNedic any chance of getting this merged in?

**Comment by peteguyatt** [comment]
I'm happy to create a separate PR for the secrets manager fix too if there is a appetite for this

**Comment by WtfJoke** [comment]
@peteguyatt the PR https://github.com/awspring/spring-cloud-aws/pull/195 contains fixes for both (secrets manager and parameter store).

Its probably better if we discuss in the new (migrated) issue 👍

**Comment by peteguyatt** [comment]
@WtfJoke Thanks for porting the fix to the new repo and fixing secrets manager while you were at it, would you mind adding me as a contributor to the unit tests/PR in general?
